libs/platform/src/lib/approval-flow/approval-flow-user-list/approval-flow-user-list.component.ts

```diff
@@ -84,27 +88,31 @@ export class ApprovalFlowUserListComponent implements AfterViewInit, OnChanges,
     /** @hidden */
     _displayUsers: ApprovalUser[] = [];
 
+    /** @hidden */
+    _allSelectedUsers: ApprovalUser[] = [];
+
     /** @hidden */
     private _intervalID?: number;
 
+    /** @hidden */
+    private _destroyed$ = inject(DestroyedService);
+
+    /** @hidden */
+    private _dontCallSelection = false;
+
     /** @hidden */
     constructor(private _cdr: ChangeDetectorRef) {}
 
     /** @hidden */
     ngAfterViewInit(): void {
         if (this.selectedUsers.length) {
-            const selectedApproversNames = this.selectedUsers.map((approver) => approver.name);
-
-            this._selectedItems = this.listItems.filter(
-                (item) => !!item.avatar?.ariaLabel && selectedApproversNames.includes(item.avatar.ariaLabel)
-            );
```

Comment:
This is not unique which can result in multiple selection on users having same name.